### PR TITLE
[Client encryption]: Update MDE dependency to v1.2.0

### PR DESCRIPTION
**Pull request by juraj-blazek**
# Pull Request Template

## Description

Updates the `Microsoft.Data.Encryption.Cryptography` dependency to version 1.2.0. This unblocks the release of stable version of the `Microsoft.Azure.Cosmos.Encryption.Custom` package.
See also [the release notes](https://github.com/Azure/microsoft-data-encryption-sdk/blob/main/net/release-notes/README.md) for more information about the update.

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## Closing issues

Unblocks the closure of an issue: #4675

### CODE REVIEW

**Review by kirankumarkolli — APPROVED** [review]

**Comment by kirankumarkolli** [comment]
@kr-santosh, @vipulvishal-ms can you please look into it?

File: Microsoft.Azure.Cosmos.Encryption.Custom/changelog.md [hunk 1]
```diff
@@ -3,6 +3,12 @@ Preview features are treated as a separate branch and will not be included in th
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+### [Unreleased]
+
+#### Updates
+
+- [#4673]: Updates `Microsoft.Data.Encryption.Cryptography` dependency to v1.1.0.
```
Comment (kr-santosh) [inline Microsoft.Azure.Cosmos.Encryption.Custom/changelog.md:10]: can you add ### <a name="1.0.0-preview08"/> [1.0.0-preview08](https://www.nuget.org/packages/Microsoft.Azure.Cosmos.Encryption.Custom/1.0.0-preview08) - 2024-XX-XX

above and bump up version in Directory.Build.props

**Review by kirankumarkolli — APPROVED** [review]

**Review by kirankumarkolli — DISMISSED** [review]

File: Microsoft.Azure.Cosmos.Encryption.Custom/changelog.md [hunk 2]
```diff
@@ -20,17 +26,20 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### <a name="1.0.0-preview04"/> [1.0.0-preview04](https://www.nuget.org/packages/Microsoft.Azure.Cosmos.Encryption.Custom/1.0.0-preview04) - 2022-08-16
 
-#### Fixes 
+#### Fixes
+
 - [#3386](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3386) Fixes custom serializer issue with DataEncryptionKeyContainer operations.
 
 ### <a name="1.0.0-preview03"/> [1.0.0-preview03](https://www.nuget.org/packages/Microsoft.Azure.Cosmos.Encryption.Custom/1.0.0-preview03) - 2022-04-15
+
```
Comment (kr-santosh) [inline Microsoft.Azure.Cosmos.Encryption.Custom/changelog.md:34]: nit: can we fix these WS/NL adds

**Review by kr-santosh — APPROVED** [review]

**Review by kr-santosh — APPROVED** [review]

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**